wire-ios-data-model/Tests/Source/Model/Conversation/ZMConversationTests+Legalhold.swift

```diff
@@ -37,17 +37,22 @@ class ZMConversationTests_Legalhold: ZMConversationTestsBase {
         }
     }
 
-    func testThatLegalholdIsDeactivatedForUser_WhenLegalholdClientIsDeleted() {
-        syncMOC.performGroupedBlock {
-            // GIVEN
-            let user = ZMUser.insertNewObject(in: self.syncMOC)
-            let legalHoldClient = self.createClient(ofType: .legalHold, class: .legalHold, for: user)
+    func testThatLegalholdIsDeactivatedForUser_WhenLegalholdClientIsDeleted() async {
+        // GIVEN
+        var legalHoldClient: UserClient!
```

Comment:
```
    func testThatLegalholdIsDeactivatedForUser_WhenLegalholdClientIsDeleted() async {
        // GIVEN
        let (user, legalHoldClient) = await syncMOC.perform {
            let user = ZMUser.insertNewObject(in: self.syncMOC)
            let legalHoldClient = self.createClient(ofType: .legalHold, class: .legalHold, for: user)
            XCTAssertTrue(user.isUnderLegalHold)
            return (user, legalHoldClient)
        }

        // WHEN
        await legalHoldClient.deleteClientAndEndSession()

        // THEN
        await syncMOC.perform {
            XCTAssertFalse(user.isUnderLegalHold)
        }
    }
```

Reply:
Yes, this is nicer but there's a ton of churn updating all tests to be like this, let's create a ticket for doing this separately?